tests/end-to-end/backup-restore-test/backupe2e/grafana.go

```diff
@@ -0,0 +1,392 @@
+// Before backup and after restore test.
+// This Test:
+// - Verify if the grafana pod is running.
+// - Login in the grafana ui.
+// - Gets Dex's credentials from Kubernetes Secret.
+// - Authenticates through Dex form.
+// - Execute http request to the grafana api to get the list of folders and dashboards.
+// - Execute a http request to every dashboard through the grafana api , expecting a 200 http status code.
+// Login and Auth (Grafana and Dex)
+// 1. https://grafana.kyma.local/login (GET)
+// 2. https://grafana.kyma.local/login/generic_oauth (GET)
+// 3. https://dex.kyma.local/auth (GET)
+// 4. https://dex.kyma.local/auth/local (POST)
+// Api:
+// 1. https://grafana.kyma.local/api/search?folderIds=0
+// 2. https://grafana.kyma.local/api/search?query=Lambda%20Dashboard
+
+package backupe2e
+
+import (
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/kubernetes"
+	"github.com/google/uuid"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+	"fmt"
+	"net/url"
+	"net/http"
+	"net/http/cookiejar"
+	"time"
+	"strings"
+	"crypto/tls"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+const (
+	grafanaNS              = "kyma-system"
+	grafanaPodName         = "monitoring-grafana-0"
+	grafanaServiceName     = "monitoring-grafana"
+	grafanaStatefulsetName = "monitoring-grafana"
+	adminUserSecretName    = "admin-user"
+	containerName          = "grafana"
+	grafanaLabelSelector   = "app=monitoring-grafana"
+)
+
+var (
+	dashboards = make(map[string]dashboard)
+)
+
+type grafanaTest struct {
+	grafanaName string
+	uuid        string
+	coreClient  *kubernetes.Clientset
+	before      bool
+	grafana
+}
+
+type grafana struct {
+	url        string
+	oauthUrl   string
+	loginForm  url.Values
+	httpClient *http.Client
+}
+
+type dashboard struct {
+	title string
+	url   string
+}
+
+func NewGrafanaTest() (*grafanaTest, error) {
+
+	kubeconfig := os.Getenv("KUBECONFIG")
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return &grafanaTest{}, err
+	}
+
+	coreClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return &grafanaTest{}, err
+	}
+
+	return &grafanaTest{
+		coreClient:  coreClient,
+		grafanaName: "grafana",
+		uuid:        uuid.New().String(),
+		before:      false,
```

Comment:
thats a bit counter intuitive.
in the before case t.before should be true.
I would also like to have it renamed do beforeBackup (or similar)
Then after it was executed before the backup you could set it to false

Reply:
That is right, should be `beforeBackup :      true` to get into the if in the first run. In the way how it is now it looks more as an `afterBackup` var.